app/controllers/providers/statement_of_cases_controller.rb

```diff
@@ -6,21 +6,19 @@ class StatementOfCasesController < ProviderBaseController
     before_action :authorize_legal_aid_application
 
     def show
-      statement_of_case
+      @form = StatementOfCases::StatementOfCaseForm.new(model: statement_of_case)
     end
 
     def update
-      if statement_of_case.update(statement_of_case_params) || draft_selected?
-        continue_or_draft
-      else
-        render :show
-      end
+      @form = StatementOfCases::StatementOfCaseForm.new(statement_of_case_params.merge(model: statement_of_case))
+
+      render :show unless save_continue_or_draft(@form)
     end
 
     private
 
     def statement_of_case_params
-      params.require(:statement_of_case).permit(:statement, :original_file)
+      params.require(:statement_of_case).permit(:statement, :original_file).merge(provider_uploader: current_provider)
```

Comment:
Is it worth just doing one `merge` on the `statement_of_case_params`. So here:
```ruby
params.require(:statement_of_case).permit(:statement, :original_file).merge(provider_uploader: current_provider, model: statement_of_case)
```
And remove the merge in the update action.